public/app/features/explore/extensions/AddToDashboard/addToDashboard.ts

```diff
@@ -28,14 +30,77 @@ function createDashboard(): DashboardDTO {
   return dto;
 }
 
+/**
+ * Returns transformations for the logs table visualisation in explore.
+ * If the logs table supports a labels column, we need to extract the fields.
+ * Then we can set the columns to show in the table via the organize/includeByName transformation
+ * @param panelType
+ * @param options
+ */
+function getLogsTableTransformations(panelType: string, options: AddPanelToDashboardOptions): DataTransformerConfig[] {
+  let transformations: DataTransformerConfig[] = [];
+  if (panelType === 'table' && options.panelState?.logs?.columns) {
+    // If we have a labels column, we need to extract the fields from it
+    if (options.panelState.logs?.labelName) {
+      transformations.push({
+        id: 'extractFields',
+        options: {
+          source: options.panelState.logs.labelName,
+        },
+      });
+    }
+
+    // Show the columns that the user selected in explore
+    transformations.push({
+      id: 'organize',
+      options: {
+        includeByName: Object.values(options.panelState.logs.columns).reduce(
+          (acc: Record<string, boolean>, value: string) => ({
+            ...acc,
+            [value]: true,
+          }),
+          {}
+        ),
+      },
+    });
+  }
+  return transformations;
+}
+
+/**
+ * Returns the transformations to apply to the panel based on the panel type
+ * @param panelType
+ * @param options
+ */
+function getExploreToPanelTransformations(panelType: string, options: AddPanelToDashboardOptions) {
+  let transformations: DataTransformerConfig[] = [];
+  transformations.push(...getLogsTableTransformations(panelType, options));
+
+  return transformations;
+}
```

Comment:
this seems to just proxy the call to `getLogsTableTransformations`, can we just use `getLogsTableTransformations` on line 103?

```suggestion
```